### ISSUE TITLE
Delete Posts by Post status - Custom Post status - Add tests

### ISSUE THREAD

**Issue by mariadanieldeepak** [sudar/bulk-delete#189]
This should be done after #217 

- [x] Create custom post status and manually verify that it is displayed in the UI
- [x] Create custom post status and manually verify that it works
- [x] Write a helper function in the test library to create posts from a custom post status
~-[ ] Write a helper function in the test library to retrieve posts from a custom post status~ Its already done by sudar `get_posts_by_status()`
- [x] Write tests to delete posts from a single custom post status
- [x] Write tests to delete posts from two custom post status
- [x] Write tests to delete posts from one custom post status and one built-in post status

Once this task is done, the helper functions should be moved to the tests library repo.

While sending this task for code review create a new issue for moving the helper functions to the the test library repo

**Comment by mariadanieldeepak** [comment]
@sudar, @rajanit2000,

I see few things to be corrected.

> https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L215

`force_delete` should be `true` since we're deleting permanently and not trashing it.

Similar changes to be made in the following lines too.

- https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L248

- https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L289

- https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L322

- https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L360

- https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L389

> https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L405

The function name should indicate if the function deletes or trashes. Also, fix the `force_delete` flag accordingly.

The following functions are missing their counterparts (i.e. Either Trashed TC is written but not deleted the vice versa)

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L163-L187

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L192-L228

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L192-L228

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L266-L302

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L307-L335

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L307-L335

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L377-L400

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L405-L423

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L428-L454

https://github.com/sudar/bulk-delete/blob/dev/6.0.0/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php#L459-L482

@rajanit2000,

Glad that you were able to write test cases. Totally appreciate it. Also, do note that the Test cases are like frontline soldiers that should take the first hit when a release is planned. So keeping them solid saves us from pain.

Also, how about naming the post statuses similar to the process in baking a Pizza? Ex. Preparing Dough, Baking, Topping. ;) Leaving some Bread crumbs in the code makes it even more delicious. What do you think?

**Comment by sudar** [comment]
@mariadanieldeepak 

Thanks for reviewing the tests and for coming up with the list of changes.

Let's create a new issue to track them instead of reopening the old one. That way we won't have any issues while naming the branch.
